test-utils/src/main/kotlin/Extensions.kt

```diff
@@ -66,3 +66,23 @@ fun TestConfiguration.createSpecTempFile(prefix: String? = null, suffix: String?
 
     return file
 }
+
+fun TestConfiguration.createTestTempDir(vararg infixes: String): File {
+    val dir = createOrtTempDir(*infixes)
+
+    afterTest {
+        dir.safeDeleteRecursively(force = true)
+    }
+
+    return dir
+}
+
+fun TestConfiguration.createTestTempFile(prefix: String? = null, suffix: String? = null): File {
```

Comment:
Why did you decide to have separate functions for 'spec' / 'test'? 
I'm asking because it would be possible to do the cleanup always consistently at the later point in time of the two, ´afterSpec` or `afterTest`. (I don't recall right now which that is)

If that's an optimization, do you think it's worth the added complexity?

Reply:
> Why did you decide to have separate functions for 'spec' / 'test'?

I didn't want to change the functionality compared to before at the same time when doing this refactoring to ease the review.

> it would be possible to do the cleanup always consistently at the later point in time

Not necessarily. Individual test might write to the same file in the temporary directory. and the second test might not expect the file to already be present from the first test.

> If that's an optimization, do you think it's worth the added complexity?

It's not a conscious optimization that I did, I just kept the existing behavior. In that sense I also did not *add* complexity.

Reply:
> Individual test might write to the same file in the temporary directory.

Convinced, missed that point.